include/reshade_api.hpp

```diff
@@ -553,11 +553,18 @@ namespace reshade::api
 			return get_preprocessor_definition(name, value, &length);
 		}
 		/// <summary>
-		/// Defines a global preprocessor definition to the specified <paramref name="value"/>.
+		/// Defines a preprocessor definition to the specified <paramref name="value"/>.
 		/// </summary>
 		/// <param name="name">Name of the definition.</param>
 		/// <param name="value">Value of the definition.</param>
 		virtual void set_preprocessor_definition(const char *name, const char *value) = 0;
+		/// <summary>
+		/// *TODO*
+		/// </summary>
+		/// <param name="effect_name">*TODO*</param>
+		/// <param name="name">Name of the definition.</param>
+		/// <param name="value">Value of the definition.</param>
+		virtual void set_preprocessor_definition(const char *effect_name, const char *name, const char *value) = 0;
```

Comment:
This will break the vtable layout and crash with any existing add-ons using the `render_technique` or below functions (which do exist already).